Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -9860,6 +9862,120 @@ void WebPage::hasActiveNowPlayingSessionChanged(bool hasActiveNowPlayingSession)
+
+    auto unobscuredContentRect = view->unobscuredContentRect();
+    auto searchRange = makeRangeSelectingNodeContents(*bodyElement);
+    while (is_lt(treeOrder<ComposedTree>(searchRange.start, searchRange.end))) {
+        auto range = findPlainText(searchRange, targetText, {
+            FindOption::CaseInsensitive,
+            FindOption::AtWordStarts,
+            FindOption::TreatMedialCapitalAsWordStart,
+            FindOption::DoNotRevealSelection,
+            FindOption::DoNotSetSelection,
+        });
+
+        if (range.collapsed())
+            break;
+
+        searchRange.start = range.end;
+
+        RefPtr<HTMLElement> target;
+        for (RefPtr ancestor = range.start.container.ptr(); ancestor; ancestor = ancestor->parentElementInComposedTree()) {
+            RefPtr element = dynamicDowncast<HTMLElement>(*ancestor);
+            if (!element)
+                continue;
+
+            if (element->willRespondToMouseClickEvents() || element->isLink()) {
+                target = WTFMove(element);
+                break;
+            }
+        }
+
+        if (!target)
+            continue;
+
+        auto textRects = RenderObject::absoluteTextRects(range, {
+            RenderObject::BoundingRectBehavior::RespectClipping,
+            RenderObject::BoundingRectBehavior::UseVisibleBounds,
+            RenderObject::BoundingRectBehavior::IgnoreTinyRects,
+            RenderObject::BoundingRectBehavior::IgnoreEmptyTextSelections,
+        });
+
+        auto indexOfFirstRelevantTextRect = textRects.findIf([&](auto& textRect) {
+            return unobscuredContentRect.intersects(textRect);
+        });
+
+        if (indexOfFirstRelevantTextRect == notFound)
+            continue;
+
+        candidates.append({ target.releaseNonNull(), textRects[indexOfFirstRelevantTextRect].center() });
+    }
+
+    candidates.removeAllMatching([&](auto& targetAndLocation) {
+        auto& [target, location] = targetAndLocation;
+        auto result = localMainFrame->eventHandler().hitTestResultAtPoint(location, {
+            HitTestRequest::Type::ReadOnly,
+            HitTestRequest::Type::Active,
+        });
+
+        RefPtr innerNode = result.innerNonSharedNode();
+        return !innerNode || !target->containsIncludingShadowDOM(innerNode.get());
+    });
+
+    if (candidates.isEmpty()) {
+        WEBPAGE_RELEASE_LOG(MouseHandling, "WebPage::simulateClickOverText - no matches found");
+        return completion(false);
+    }
+
+    if (candidates.size() >= 2) {
+        WEBPAGE_RELEASE_LOG(MouseHandling, "WebPage::simulateClickOverText - too many matches found (%zu)", candidates.size());
+        // FIXME: We'll want to add a way to disambiguate between multiple matches in the future. For now, just exit without
+        // trying to simulate a click.
+        return completion(false);
+    }
+
+    auto& [target, location] = candidates.first();
+
+    SetForScope userIsInteractingChange { m_userIsInteracting, true };
```

Comment:
It's rather subtle that this function sets this boolean to true.
Maybe this should also be reflected in the function name?